docs/DeveloperGuide.md

```diff
@@ -2,36 +2,524 @@
+  those
+  that match the specified criteria.
+
+- **Result Presentation:** The function displays the search results, presenting users with a list of items that meet the
+  search criteria, allowing them to quickly identify the items they are interested in.
+
+- **User-Friendly Interface:** The "Find" function is integrated into the user interface, making it easily accessible
+  and intuitive for users to perform searches.
+
+- **Alternative Consideration:** During the design process, alternative approaches to searching are evaluated to ensure
+  the
+  most effective and user-friendly method is implemented.
+
+The "Find" function offers a valuable way for users to narrow down their searches, find specific items of interest, and
+enhance the usability of the application.
+
+**Function Methods**
+
+The "Find" function includes the following method to achieve its functionality:
+
+- `execute()` - This method is responsible for executing the "Find" command, searching for drugs that match the
+  user-specified keyword within the inventory.
+- It returns a `CommandResult` containing the outcome of the command execution,
+  which includes a success message and a list of found StockEntry objects.
+
+**Example Usage**
+
+To illustrate how the "Find" function works, consider the following example usage:
+
+1. **User Input:** The user initiates the "Find" command by typing something like the following:
+
+`find /n panadol` - This command instructs the program to search for drugs in the inventory based on the name criteria
+and the keyword "panadol."
+
+`find /d 12/03/2026` - This command instructs the program to search for drugs in the inventory based on the expiry date
+criteria and the keyword "12/03/2026."
+
+`find /s PARC3189` - This command instructs the program to search for drugs in the inventory based on the serial number
+criteria and the keyword "PARC3189."
+
+2. **Method Execution:** The `execute()` method within the "FindCommand" class is called. It takes the provided keyword
+   and criterion as input.
+
+3. **Search Process:** The method processes the search by iterating through the list of `StockEntry` objects in the
+   inventory.
+   For each entry, it checks if the `matches` method returns `true`, which means that the drug name or expiry date
+   contains the given keyword.
+4. **Building Results:** As matching entries are found, they are added to a list called `foundEntries`.
+5. **Result Display:** The `CommandResult` is prepared, containing a success message (e.g., "Listed all drugs with the
+   keyword
+   in the inventory.") and the list of found `StockEntry` objects.
+6. `User Feedback:` The result is then displayed to the user, showing a list of drugs in the inventory that match the
+   specified keyword.
+
+The following sequence diagram shows how the Find Command function works.
+
+<img src="UML Diagrams/FindCommandDiagram.png" width="350">
```

Comment:
Are the activation bars meant to terminate at that point? Or is the diagram cut off / incomplete?